src/controllers/footage.controller.ts

```diff
@@ -158,10 +170,74 @@ const getFootageClips = async (
   return res.status(200).json({ data: footageClips });
 };
 
-// TODO: Implement updateFootage endpoint to update after parsing & analysis.
-// const updateFootage = async (req: Request, res: Response) => {
-//
-// };
+/**
+ * PATCH /footage/:uuid
+ * @summary Endpoint to update a specific Footage document.
+ * @return {string, FootageDocument} 200 - Success response returns the Footage document updated and a message.
+ * @return 400 - Fields isCsgoFootage & isAnalyzed were not provided.
+ * @return 406 - One of the params (UUID, isCsgoFootage or isAnalyzed) was not provided.
+ * @return 412 - No document with the provided UUID was found.
+ * @return 418 - An error occured while attempting to update the FootageDocument.
+ */
+const updateFootage = async (
+  req: Request,
+  res: Response,
+): Promise<Response<any, Record<string, any>>> => {
+  const { uuid } = req.params;
+  const { isCsgoFootage, isAnalyzed } = req.body;
+  // check if all fields were supplied
+  if (
+    uuid === undefined ||
+    isCsgoFootage === undefined ||
+    isAnalyzed === undefined
+  ) {
+    return res.status(400).json({
+      message: 'The fields isCsgoFootage and isAnalyed must be supplied.',
+    });
+  }
+
+  // check if id provided is a UUIDv4.
+  if (!validate(uuid)) {
+    return res
+      .status(406)
+      .json({ message: `UUID Param: ${uuid} is not a valid UUID.` });
+  }
+  // username is type of string?
+  // isCsgoFootage is type of boolean?
+  if (typeof isCsgoFootage !== 'boolean') {
+    return res
+      .status(406)
+      .json({ message: 'The isCsgoFootage field must be a boolean.' });
+  }
+  // isAnalyzed is type of boolean?
+  if (typeof isAnalyzed !== 'boolean') {
+    return res
+      .status(406)
+      .json({ message: 'The isAnalyzed field must be a boolean.' });
+  }
```

Comment:
Doesn't need to be done now, but give how much this is all repeated, we should probably have make util funcs for all this so we don't have a bunch of duplicate code. We can also consider using zod for this, and I think it can do most if not all of this.